### DISCUSSION TITLE
Register recipe but result doesn't appear

### DISCUSSION THREAD

**Discussion by Distractic** [Minestom/Minestom#1663] [Help]
I'm looking to register recipe. I checked #636 #1134 to implement my first crafting table recipe.

So like #1134 , I created the class  : 

```kotlin
class CraftingTable : ShapedRecipe(
    Material.CRAFTING_TABLE.namespace().asString(), 2, 2, "",
    listOf(
        ingredients, ingredients,
        ingredients, ingredients
    ),
    ItemStack.of(Material.CRAFTING_TABLE)
) {

    companion object {
        private val ingredients: DeclareRecipesPacket.Ingredient = DeclareRecipesPacket.Ingredient(
            listOf(
                ItemStack.of(Material.OAK_PLANKS),
                ItemStack.of(Material.ACACIA_PLANKS),
                ItemStack.of(Material.BIRCH_PLANKS),
                ItemStack.of(Material.JUNGLE_PLANKS),
                ItemStack.of(Material.SPRUCE_PLANKS),
                ItemStack.of(Material.DARK_OAK_PLANKS),
                ItemStack.of(Material.WARPED_PLANKS),
                ItemStack.of(Material.CRIMSON_PLANKS)
            )
        )
    }

    override fun shouldShow(player: Player): Boolean = true

}
```

And in my main class, I'm registering the recipe using `MinecraftServer.getRecipeManager().addRecipe(CraftingTable())`

However, in game, the crafting table doesn't appear in result item
![image](https://user-images.githubusercontent.com/46402441/210975304-eb1753f1-590f-41c7-af1d-2c5189115bc8.png)

Can you help me to fix that ?

**Comment by Flopgop** [comment]
You have to process the recipe yourself somehow if I recall correctly

**Reply by Distractic** [reply]
Do you have any clue about how to do that ?

**Comment by Distractic** [comment]
In the crafting table, the recipe is registered, but I can't process it, so like @Flopgop say, I need to process the recipe myself, but I have no idea how I can do that

![image](https://user-images.githubusercontent.com/46402441/210977581-aa01859f-3346-4d92-b574-561a88e559e6.png)
